File: bindings/ruby/test/test_base.rb
require 'minitest/autorun'
require_relative '../lib/cconfigspace'

class CConfigSpaceTest < Minitest::Test
  def test_version
    ver = CCS.version
    assert(ver.kind_of?(CCS::Version))
  end

  def test_version_string
    assert_output(/^v\d+\.\d+\.\d+\.\d+/) { puts CCS.version_string }
  end

  def test_datum_value
    d = CCS::Datum::new
    d[:type] = :CCS_DATA_TYPE_NONE
    assert_nil( d.value )
    d[:type] = :CCS_DATA_TYPE_INACTIVE
    assert_equal( CCS::Inactive, d.value )
    d[:type] = :CCS_DATA_TYPE_BOOL
    d[:value][:i] = CCS::FALSE
    assert_equal( false, d.value )
    d[:type] = :CCS_DATA_TYPE_BOOL
    d[:value][:i] = CCS::TRUE
    assert_equal( true, d.value )
    d[:type] = :CCS_DATA_TYPE_FLOAT
    d[:value][:f] = 15.0
    assert_equal( 15.0, d.value )
    d[:type] = :CCS_DATA_TYPE_INT
    d[:value][:i] = 15
    assert_equal( 15, d.value )
    ptr = CCS::MemoryPointer::from_string("foo")
    d[:type] = :CCS_DATA_TYPE_STRING
    d[:value][:s] = ptr
    assert_equal( "foo", d.value )
    rng = CCS::Rng::new
    d[:type] = :CCS_DATA_TYPE_OBJECT
    d[:value][:o] = rng.handle
    o = d.value
    assert( o.kind_of? CCS::Rng )
    assert_equal( :CCS_OBJECT_TYPE_RNG, d.value.object_type )
  end

  def test_from_value
    d = CCS::Datum::from_value(nil)
    assert_equal( :CCS_DATA_TYPE_NONE, d[:type] )
    assert_equal( 0, d[:value][:i] )
    d = CCS::Datum::from_value(CCS::Inactive)
    assert_equal( :CCS_DATA_TYPE_INACTIVE, d[:type] )
    assert_equal( 0, d[:value][:i] )
    d = CCS::Datum::from_value(false)
    assert_equal( :CCS_DATA_TYPE_BOOL, d[:type] )
    assert_equal( CCS::FALSE, d[:value][:i] )
    d = CCS::Datum::from_value(true)
    assert_equal( :CCS_DATA_TYPE_BOOL, d[:type] )
    assert_equal( CCS::TRUE, d[:value][:i] )
    d = CCS::Datum::from_value(15)
    assert_equal( :CCS_DATA_TYPE_INT, d[:type] )
    assert_equal( 15, d[:value][:i] )
    d = CCS::Datum::from_value(15.0)
    assert_equal( :CCS_DATA_TYPE_FLOAT, d[:type] )
    assert_equal( 15.0, d[:value][:f] )
    rng = CCS::Rng::new
    d = CCS::Datum::from_value(rng)
    assert_equal( :CCS_DATA_TYPE_OBJECT, d[:type] )
    assert_equal( rng.handle, d[:value][:o] )
  end

  def test_value_affect
    d = CCS::Datum::from_value(nil)
    assert_equal( :CCS_DATA_TYPE_NONE, d[:type] )
    assert_equal( 0, d[:value][:i] )
    d.value = CCS::Inactive
    assert_equal( :CCS_DATA_TYPE_INACTIVE, d[:type] )
    assert_equal( 0, d[:value][:i] )
    d.value = false
    assert_equal( :CCS_DATA_TYPE_BOOL, d[:type] )
    assert_equal( CCS::FALSE, d[:value][:i] )
    d.value = true
    assert_equal( :CCS_DATA_TYPE_BOOL, d[:type] )
    assert_equal( CCS::TRUE, d[:value][:i] )
    d.value = 15
    assert_equal( :CCS_DATA_TYPE_INT, d[:type] )
    assert_equal( 15, d[:value][:i] )
    d.value = 15.0
    assert_equal( :CCS_DATA_TYPE_FLOAT, d[:type] )
    assert_equal( 15.0, d[:value][:f] )
    rng = CCS::Rng::new
    d.value = rng
    assert_equal( :CCS_DATA_TYPE_OBJECT, d[:type] )
    assert_equal( rng.handle, d[:value][:o] )
    d.value = nil
    assert_equal( :CCS_DATA_TYPE_NONE, d[:type] )
    assert_equal( 0, d[:value][:i] )
  end

  def test_numeric
    n = CCS::Numeric::from_value(1)
    assert_equal( 1, n.value(:CCS_NUMERIC_TYPE_INT) )
    n.value = 2
    assert_equal( 2, n.value(:CCS_NUMERIC_TYPE_INT) )
    n.value = 1.0
    assert_equal( 1.0, n.value(:CCS_NUMERIC_TYPE_FLOAT) )
    n = CCS::Numeric::from_value(2.0)
    assert_equal( 2.0, n.value(:CCS_NUMERIC_TYPE_FLOAT) )
  end
end

File: bindings/python/test/test_base.py
import unittest
import sys
sys.path.insert(1, '.')
sys.path.insert(1, '..')
import cconfigspace as ccs
from math import sin


class TestBase(unittest.TestCase):

  def test_version(self):
    ver = ccs.ccs_get_version()
    self.assertIsInstance(ver, ccs.Version)

  def test_datum_value(self):
    d = ccs.Datum()
    d.value = None
    self.assertEqual( ccs.DataType.NONE, d.type )
    self.assertIsNone( d.value )
    d.value = ccs.inactive
    self.assertEqual( ccs.DataType.INACTIVE, d.type )
    self.assertEqual( ccs.inactive, d.value )
    d.value = False
    self.assertEqual( ccs.DataType.BOOL, d.type )
    self.assertFalse( d.value )
    d.value = True
    self.assertEqual( ccs.DataType.BOOL, d.type )
    self.assertTrue( d.value )
    d.value = 1.5
    self.assertEqual( ccs.DataType.FLOAT, d.type )
    self.assertEqual( 1.5, d.value )
    d.value = 2
    self.assertEqual( ccs.DataType.INT, d.type )
    self.assertEqual( 2, d.value )
    d.value = "foo"
    self.assertEqual( ccs.DataType.STRING, d.type )
    self.assertEqual( "foo", d.value )
    d.value = ccs.Rng()
    self.assertEqual( ccs.DataType.OBJECT, d.type )
    self.assertIsInstance( d.value, ccs.Rng )
    self.assertEqual( ccs.ObjectType.RNG, d.value.object_type )

  def test_numeric(self):
    n = ccs.Numeric()
    n.set_value(2)
    self.assertEqual( 2, n.get_value(ccs.NumericType.INT))
    n.set_value(1.5)
    self.assertEqual( 1.5, n.get_value(ccs.NumericType.FLOAT))

if __name__ == '__main__':
    unittest.main()
